spec/index.bs

```diff
@@ -1409,62 +1409,63 @@ interface GPUBufferUsage {
 };
 </script>
 
-### <dfn method for=GPUDevice>createBuffer(descriptor)</dfn> ### {#GPUDevice-createBuffer}
-
-<div algorithm="GPUDevice.createBuffer">
-  **Arguments:**
-    - {{GPUBufferDescriptor}} |descriptor|
+<dl dfn-type=method dfn-for=GPUDevice>
+    : <dfn>createBuffer(descriptor)</dfn>
+    ::
 
-  **Returns:** {{GPUBuffer}}
+    <div algorithm=GPUDevice.createBuffer>
+        **Called on:** {{GPUDevice}} |this|.
 
-  1. If this call doesn't follow the [$createBuffer Valid Usage$]:
+        **Arguments:**
+        <pre class=argumentdef for="GPUDevice/createBuffer(descriptor)">
+            |descriptor|:
+        </pre>
 
-    1. Retun an error buffer.
+        **Returns:** {{GPUBuffer}}
 
-    Issue(gpuweb/gpuweb#605): Explain that the resulting error buffer can still be mapped at creation.
+        - If any of the following conditions are unsatisfied, return an error buffer and stop.
+            <div class=validusage>
+                - |this| is a [=valid=] {{GPUDevice}}.
+                - |descriptor|.{{GPUBufferDescriptor/usage}} is a subset of |this|.[[allowed buffer usages]].
+                - If |descriptor|.{{GPUBufferDescriptor/usage}} contains {{GPUBufferUsage/MAP_READ}} then
+                    the only other usage it may contain is {{GPUBufferUsage/COPY_DST}}.
```

Comment:
The "may contain" language is a little inconsistent with the "is [a valid device]" language. From your experience editing these do you think we should make it look like:

- If descriptor.usage contains MAP_READ, then:
    - descriptor.usage does not contain any other usages except COPY_DST

or similar, or change how we write the language here, or keep this?

Reply:
Good point, though it feels slightly awkward no matter how it's phrased to me. Tried something that's a close variant on what you suggested, let me know how it feels to you!

Reply:
Looks good, but yeah, I don't think there's any way to make it less awkward with this phrasing. I suppose if we switched it all back to "must" style language it might read clearer, but I'm not sure making that global change is a good idea.